tools/ci/osquery-ubuntu18.04-toolchain.dockerfile

```diff
@@ -45,6 +44,19 @@ RUN case $(uname -m) in aarch64) ARCH="aarch64" ;; amd64|x86_64) ARCH="x86_64" ;
 FROM base2 AS base3
 RUN locale-gen en_US.UTF-8
 
+FROM base3 AS cppcheck
+ENV cppcheckVer 2.6.3
+WORKDIR /root
+RUN case $(uname -m) in amd64|x86_64) git clone https://github.com/danmar/cppcheck.git \
```

Comment:
While I can live wit this, it is unnecessary (and IMO confusing). The docker pattern of creating large `RUN` lines is to avoid bloating the final image with unneeded layers. 

But we achieve that in the final `FROM scratch AS builder` anyhow.

Reply:
But here I need to conditionally run the operations, so  the long list of commands has to run on amd64, while the very last mkdir is for arm64 and not fail the `COPY` operation later.

Are you suggesting to separate each command on it's own `RUN` and put an `if` everytime to check if the platform is amd64?
Wouldn't that be even more noisy?

Reply:
Oh, I didn't even notice the `if`. I don't know if there's anything cleaner. 

> cppcheck step and the cleaning of the apt database

If you do the layers right, you don't need to clean the apt db.

Reply:
> If you do the layers right, you don't need to clean the apt db.

How do you mean?
The only way I know to leave things behind is to do a separate `FROM x AS y` as we've been doing and then copy portions from it, but stuff installed from apt can land in various folder, I don't know if it's worth to try to piece things like that in the final step?

Reply:
Yes, I think so. Still from apt can get sprayed everywhere, and it's hard to clean up. But that doesn't matter if you're throwing the whole container away and only doing `COPY --from=cppcheck /root/cppcheck/install/usr/local/ /usr/local/`

I think we can probably keep this, but I find it hard to read.

Reply:
I think we're talking cross purposes, or mostly agreeing, or something like that. 

I'm observing that the length of the RUN command in the `cppcheck` container makes it hard to follow, and that since we're going to throw most of this container away later, we can use multiple RUN lines. 

You, reasonable, pointed out that it's a mechanism to switch on platform. And possible the simplest.

For cleaning up the apt db stuff, if we wanted to toss any `rm` things in into `base3` that seems reasonable. I don't think we need to have a working apt system at the end of this,